File: 0724-find-pivot-index/0724-find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int totalSum = 0;
        int sumLeft = 0;
      
        for (int num: nums) {
          totalSum += num;
        }
      
        for (int i = 0; i < nums.length; i++) {
            int sumRight = totalSum - sumLeft - nums[i];

            if (sumLeft == sumRight) {
                return i;
            } else {
                sumLeft += nums[i];
            }
        }

        System.out.println(totalSum);

        return -1;
    }
}

File: 0724-find-pivot-index/0724-find-pivot-index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
  let max = 0;
  let leftSum = 0;
  let rightSum = 0;
  
  for (let num of nums) max += num;
  for (let i = 0; i < nums.length; i ++) {
    rightSum = max - leftSum - nums[i];
    
    if (leftSum === rightSum) return i;
    
    leftSum += nums[i];
  }
  
  return -1;
}

var pivotIndexLoop = function(nums) {
  for (let i = 1; i < nums.length; i++) {
    nums[i] += nums[i - 1];
  }
  
  let max = nums[nums.length - 1];
  let leftSum = 0;
  let rightSum = 0;
  
  for (let i = 0; i < nums.length; i++) {
    rightSum = max - nums[i];
    
    if (leftSum === rightSum) return i;
    
    leftSum = nums[i];
  }
    
  return -1;
};